controllers/dockermachine_controller.go

```diff
@@ -72,17 +75,46 @@ func (r *DockerMachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 		return ctrl.Result{}, nil
 	}
 
-	// Get the Cluster API Cluster
+	log = log.WithValues("machine", machine.Name)
+
+	// Fetch the Cluster.
 	cluster, err := util.GetClusterFromMetadata(ctx, r.Client, machine.ObjectMeta)
 	if err != nil {
-		log.Error(err, "Failed to get cluster for docker machine")
+		log.Info("Machine is missing cluster label or cluster does not exist")
 		return ctrl.Result{}, err
 	}
 	if cluster == nil {
 		log.Info(fmt.Sprintf("Please associate this machine with a cluster using the label %s: <name of cluster>", clusterv1.MachineClusterLabelName))
 		return ctrl.Result{}, nil
 	}
 
+	log = log.WithValues("cluster", cluster.Name)
+
+	// Fetch the Docker Cluster.
+	dockerCluster := &infrav1.DockerCluster{}
+	dockerClusterName := types.NamespacedName{
+		Namespace: dockerMachine.Namespace,
+		Name:      cluster.Spec.InfrastructureRef.Name,
+	}
+	if err := r.Client.Get(ctx, dockerClusterName, dockerCluster); err != nil {
+		log.Info("Waiting for DockerCluster")
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
```

Comment:
agreed, since docker machines now depend on docker clusters we should add a watch for DockerCluster that triggers a reconcile of the associated DockerMachines.